src/systems/force_torque/ForceTorque.cc

```diff
@@ -0,0 +1,347 @@
+                 << "] not found. Failed to create sensor [" << sensorScopedName
+                 << "]" << std::endl;
+          return true;
+        }
+
+        const auto jointChildName =
+            _ecm.Component<components::ChildLinkName>(jointEntity)->Data();
+        auto jointChildLinkEntity =
+            this->GetLinkFromScopedName(_ecm, jointChildName, modelEntity);
+        if (kNullEntity == jointChildLinkEntity)
+        {
+          ignerr << "Child link with name [" << jointChildName
+                 << "] of joint with name [" << jointName
+                 << "] not found. Failed to create sensor [" << sensorScopedName
+                 << "]" << std::endl;
+          return true;
+        }
+
+        SensorJointAndLinks sensorJointLinkEntry;
+        sensorJointLinkEntry.joint = jointEntity;
+        sensorJointLinkEntry.jointParentLink = jointParentLinkEntity;
+        sensorJointLinkEntry.jointChildLink = jointChildLinkEntity;
+        this->sensorJointLinkMap[_entity] = sensorJointLinkEntry;
+
+        this->entitySensorMap.insert(
+            std::make_pair(_entity, std::move(sensor)));
+
+        return true;
+      });
+}
+
+//////////////////////////////////////////////////
+void ForceTorquePrivate::Update(const EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("ForceTorquePrivate::Update");
+  _ecm.Each<components::ForceTorque>(
+      [&](const Entity &_entity, const components::ForceTorque *) -> bool
+      {
+        auto it = this->entitySensorMap.find(_entity);
+        if (it != this->entitySensorMap.end())
+        {
+          auto jointLinkIt = this->sensorJointLinkMap.find(_entity);
+          if (jointLinkIt == this->sensorJointLinkMap.end())
+          {
+            ignerr << "Failed to update Force/Torque Sensor: " << _entity
+                   << ". Associated entities not found." << std::endl;
+            return true;
+          }
+
+          auto jointWrench = _ecm.Component<components::JointTransmittedWrench>(
+              jointLinkIt->second.joint);
+
+          // Notation:
+          // X_WJ: Pose of joint in world
+          // X_WP: Pose of parent link in world
+          // X_WC: Pose of child link in world
+          // X_WS: Pose of sensor in world
+          //
+          const auto X_WP =
+              worldPose(jointLinkIt->second.jointParentLink, _ecm);
+          const auto X_WC = worldPose(jointLinkIt->second.jointChildLink, _ecm);
+          // There appears to be a bug worldPose for computing poses of //joint
+          // and its children, so we do it manually here.
+          const auto X_CJ =
+              _ecm.Component<components::Pose>(jointLinkIt->second.joint)
+                  ->Data();
+          auto X_WJ = X_WC * X_CJ;
+
+          auto X_JS = _ecm.Component<components::Pose>(_entity)->Data();
+          auto X_WS = X_WJ * X_JS;
+          auto X_SP = X_WS.Inverse() * X_WP;
+          auto X_SC = X_WS.Inverse() * X_WC;
```

Comment:
These could be documented above too, I believe it's

```
          // X_SP: Pose of parent link in sensor frame
          // X_SC: Pose of child link in sensor frame
```

Reply:
54793e92